src/Data/Validation/Primitives.hs

```diff
@@ -126,14 +126,28 @@ nest attr validator = mapErrors (nestErrors attr) `mapResult` validator
 mustBeNull :: Validator ()
 mustBeNull = Validator $ \input ->
   case inputNull input of
-    Just _ -> Valid ()
-    Nothing -> Invalid (errMessage "must_be_null")
+    IsNull -> Valid ()
+    NotNull -> Invalid (errMessage "must_be_null")
+    InvalidNull text -> Invalid (errMessage text)
 
-nullable :: Validator a -> Validator (Maybe a)
-nullable validator =
+canBeNull :: Validator ()
+canBeNull = Validator $ \input ->
+  case inputNull input of
+    IsNull -> Valid ()
+    NotNull -> Valid ()
+    InvalidNull text -> Invalid (errMessage text)
```

Comment:
Isn't `InvalidNull text` unreachable? `inputNull` only returns values `IsNull` and `NotNull`?
Or is it there just to go through all values of `CanNull` even though it's not possible?

Reply:
On JSONs, that is true, but InvalidNull is for XML. Since it can't hold null, calling inputNull on an XML returns InvalidNull "XML_cannot_hold_null". The text is just to tell it why it's invalid.